### DISCUSSION TITLE
Sub directory

### DISCUSSION THREAD

**Discussion by ramazulay** [D10S0VSkY-OSS/Stack-Lifecycle-Deployment#165] [General]
Hi @D10S0VSkY-OSS,
I have 2 folders in the same git repo.
I need to run terraform apply in each sub folder.
Is it possible to define a subdirectory when I run deploy?

Thanks.

**Comment by D10S0VSkY-OSS** [comment]
Hi @ramazulay ,
Yes, you can do it by specifying the project_path
Regards

**Comment by ramazulay** [comment]
Great!
I didn't see the project_path value, can you guide me where should I set it?
perhaps my versions are too old:
sld-api:2.5.0
sld-dashboard:2.4.1
sld-schedule:2.2.0
sld-remote-state:2.4.0
rabbitmq:3.8-management
redis

Please let me know if I need to upgrade, if so specify the tags in any microservices.

Thanks a lot.

**Comment by D10S0VSkY-OSS** [comment]
Hi @ramazulay,
You should update to these versions:
sld-api 2.17.0
sld-dashboard dashboard 2.15.0
sld-schedule 2.6.0
sld-remote-state 2.10.0

**Important: The database schema has evolved, so you need to back up your data first. The new schema is as follows:**
```
Table: deploy
+--------------+--------------+------+-----+---------+----------------+
| Field | Type | Null | Key | Default | Extra |
+--------------+--------------+------+-----+---------+----------------+
| id | int | NO | PRI | NULL | auto_increment |
| task_id | varchar(36) | YES | | NULL | |
| name | varchar(100) | YES | | NULL | |
| action | varchar(100) | YES | | NULL | |
| start_time | varchar(100) | YES | | NULL | |
| destroy_time | varchar(100) | YES | | NULL | |
| stack_name | varchar(100) | YES | MUL | NULL | |
| stack_branch | varchar(100) | YES | | NULL | |
| created_at | datetime | YES | | NULL | |
| updated_at | datetime | YES | | NULL | |
| user_id | int | YES | | NULL | |
| username | varchar(50) | NO | | NULL | |
| squad | varchar(50) | NO | MUL | NULL | |
| variables | json | YES | | NULL | |
| environment | varchar(50) | YES | | NULL | |
| tfvar_file | varchar(50) | YES | | NULL | |
| project_path | varchar(50) | YES | | NULL | |
+--------------+--------------+------+-----+---------+----------------+
```
```
Table: stacks
+--------------+--------------+------+-----+---------+----------------+
| Field | Type | Null | Key | Default | Extra |
+--------------+--------------+------+-----+---------+----------------+
| id | int | NO | PRI | NULL | auto_increment |
| stack_name | varchar(50) | YES | UNI | NULL | |
| git_repo | varchar(200) | YES | | NULL | |
| branch | varchar(50) | YES | | NULL | |
| task_id | varchar(200) | YES | | NULL | |
| var_json | json | YES | | NULL | |
| var_list | json | YES | | NULL | |
| squad_access | json | YES | | NULL | |
| tf_version | varchar(30) | YES | | NULL | |
| project_path | varchar(50) | YES | | NULL | |
| created_at | datetime | YES | | NULL | |
| description | text | YES | | NULL | |
| user_id | int | YES | | NULL | |
+--------------+--------------+------+-----+---------+----------------+
```
To add the column "project_path" if it does not exist in the "deploy" and "stacks" tables in MySQL, you can use the following SQL statements:

For the "deploy" table:

```
ALTER TABLE deploy
ADD COLUMN IF NOT EXISTS project_path varchar(50) DEFAULT NULL;
```
For the "stacks" table
```
ALTER TABLE stacks
ADD COLUMN IF NOT EXISTS project_path varchar(50) DEFAULT NULL;
```
These statements will add the "project_path" column to the tables if it doesn't already exist. The column will have a data type of varchar(50) and a default value of NULL.

**Comment by ramazulay** [comment]
Thanks, I created a new deployment and will move the workload

**Comment by ramazulay** [comment]
I tried to set the remote state on Azure, but is still on local.
![image](https://github.com/D10S0VSkY-OSS/Stack-Lifecycle-Deployment/assets/41533452/e55e3e17-200d-4d27-9b62-532313384705)
I configured as you guide:
- name: SLD_STORE
           value: azure
- name: AZURE_STORAGE_CONNECTION_STRING
            value: "DefaultEndpointsProtocol=https;AccountName=sldtfstate;AccountKey=XXXXXXXXXXXXXX;EndpointSuffix=core.windows.net"

Can you assist please?

**Comment by D10S0VSkY-OSS** [comment]
Hi @ramazulay 
you need to set up the following variables:
SLD_STORAGE_BACKEND="azure"
SLD_CONTAINER by default is "sld-remote-state"
AZURE_STORAGE_CONNECTION_STRING=<connection_string>
Regards

**Comment by ramazulay** [comment]
Hi @D10S0VSkY-OSS ,
Thanks it works.
Please add the SLD_STORAGE_BACKEND="azure" env variable to the DOC(its not appears in your doc)

Thanks.
